sdk/tests/deqp/framework/common/tcuTestCase.js

```diff
@@ -161,9 +161,50 @@ goog.scope(function() {
     tcuTestCase.DeqpTest.prototype.init = function() {};
 
     /**
-     * Abstract deinit function(each particular test will implement it, or not)
+     * Deinit
      */
-    tcuTestCase.DeqpTest.prototype.deinit = function() {};
+    tcuTestCase.DeqpTest.prototype.deinit = function() {
+        // Multisampling state
+        gl.disable(gl.SAMPLE_COVERAGE);
+        gl.disable(gl.SAMPLE_ALPHA_TO_COVERAGE);
+
+        // Texture state
+        gl.bindTexture(gl.TEXTURE_2D, null);
```

Comment:
Note that we need to be careful not to slow down test execution too much with these state resets.